step1/vanila-todo/TodoController.ts

```diff
@@ -0,0 +1,72 @@
+import TodoModel from "./TodoModel";
+import TodoView from "./TodoView";
+
+export default class TodoController {
+  model: TodoModel;
+  view: TodoView;
+
+  constructor(model: TodoModel) {
+    this.model = model;
+    this.view = model.view;
+  }
+
+  registerEventListener() {
+    this.view.target.addEventListener("click", (e: any) => {
+      if (e.target.classList.contains("register")) {
+        this.AddTodoListData();
+      }
+      if (e.target.classList.contains("edit")) {
+        const li = e.target.closest("li");
+        const id = parseInt(li.dataset.key);
+        this.model.toggleEditMode(id);
+      }
+      if (e.target.classList.contains("delete")) {
+        const li = e.target.closest("li");
+        this.DeleteTodoListData(li);
+      }
```

Comment:
```suggestion
      const { target } = this;
      if (target.classList.contains(ButtonTypes.REGISTER)) {
        this.AddTodoListData();
      }
      if (target.classList.contains(ButtonTypes.EDIT)) {
        const li = target.closest("li");
        const id = parseInt(li.dataset.key);
        this.model.toggleEditMode(id);
      }
      if (target.classList.contains(ButtonTypes.DELETE)) {
        const li = target.closest("li");
        this.DeleteTodoListData(li);
      }
```
- 이렇게 구조해체할당을 하여 사용해주면 좋을 것 같아요 ㅎㅎ
- 그리고 class 이름을 단순하게 문자열로 사용하기 보단 ButtonTypes 라는 `enum`을 만들어서 관리해주면 더 좋을 것 같아요!

참고링크: https://www.typescriptlang.org/ko/docs/handbook/enums.html